tests/autorestart_test/test_container_autorestart.py

```diff
@@ -0,0 +1,276 @@
+                    .format(program_name, container_name, program_status))
+    else:
+        pytest.fail("Failed to find program '{}' in container '{}'"
+                    .format(program_name, container_name))
+
+    logger.info("Waiting to ensure container '{}' does not stop...".format(container_name))
+    stopped = wait_until(CONTAINER_STOP_THRESHOLD_SECS,
+                         CONTAINER_CHECK_INTERVAL_SECS,
+                         check_container_state, duthost, container_name, False)
+    pytest_assert(not stopped, "Container '{}' was stopped unexpectedly".format(container_name))
+    logger.info("Container '{}' did not stop".format(container_name))
+    logger.info("Restart the program '{}' in container '{}'".format(program_name, container_name))
+    duthost.shell("docker exec {} supervisorctl start {}".format(container_name, program_name))
+
+
+def test_containers_autorestart(duthost):
+    """
+    @summary: Test the auto-restart feature of each container against two scenarios: killing
+              a non-critical process to verify the container is still running; killing each
+              critical process to verify the container will be stopped and restarted
+    """
+    container_autorestart_states = duthost.get_container_autorestart_states()
+    disabled_containers = get_disabled_container_list(duthost)
+
+    for container_name in container_autorestart_states.keys():
+        # Skip testing the database container or containers/services which are disabled
+        if container_name in disabled_containers or container_name == "database":
+            logger.warning("Skip testing the container '{}'".format(container_name))
+            continue
+
+        is_running = is_container_running(duthost, container_name)
+        if not is_running:
+            pytest.fail("Container '{}' is not running. Exiting...".format(container_name))
+
+        logger.info("Start testing the container '{}'...".format(container_name))
+
+        restore_disabled_state = False
+        if container_autorestart_states[container_name] == "disabled":
+            logger.info("Change auto-restart state of container '{}' to be 'enabled'".format(container_name))
+            duthost.shell("config container feature autorestart {} enabled".format(container_name))
+            restore_disabled_state = True
+
+        # Currently we select 'rsyslogd' as non-critical processes for testing based on
+        # the assumption that every container has an 'rsyslogd' process running and it is not
+        # considered to be a critical process
+        program_status, program_pid = get_program_info(duthost, container_name, "rsyslogd")
+        verify_no_autorestart_with_non_critical_process(duthost, container_name, "rsyslogd",
+                                                        program_status, program_pid)
+
+        critical_group_list, critical_process_list, succeeded = duthost.get_critical_group_and_process_list(container_name)
```

Comment:
I left a comment on https://github.com/Azure/sonic-mgmt/pull/2193 asking to add an "s" to the end of `get_critical_group_and_process_list`

Reply:
Updated!